src/Diagnostics.DataProviders/DataProviders/KustoDataProvider.cs

```diff
@@ -57,6 +57,23 @@ public KustoDataProvider(OperationDataCache cache, KustoDataProviderConfiguratio
 
         public async Task<DataTable> ExecuteClusterQuery(string query, string requestId = null, string operationName = null)
         {
+            var matches = Regex.Matches(query, @"cluster\((?<cluster>([^\)]+))\).database\((?<database>([^\)]+))\)\.");
+            if (matches.Any())
+            {
+                foreach (Match element in matches)
+                {
+                    string targetCluster = element.Groups["cluster"].Value.Trim(new char[] { '\'', '\"' });
+                    string targetDatabase = element.Groups["database"].Value.Trim(new char[] { '\'', '\"' });                    
+
+                    if (!string.IsNullOrWhiteSpace(targetCluster) && !string.IsNullOrWhiteSpace(targetDatabase)
+                        && targetCluster.Equals("Aks", StringComparison.OrdinalIgnoreCase) && targetDatabase.Equals("AKSprod", StringComparison.OrdinalIgnoreCase))
+                    {
+                        //Apply this only for Aks at the moment.
+                        return await ExecuteClusterQuery(query, targetCluster, targetCluster, requestId, operationName);
```

Comment:
So if the query has a join where there are two statements with cluster and database and table information eg, <cluster>.<database>.<table> then we are executing the query twice?

Reply:
Not really, the return along with the wait will cause the function to terminate as soon as the first match is found.